core/src/main/java/discord4j/core/object/entity/User.java

```diff
@@ -97,19 +108,37 @@ public final String getUsername() {
      * Gets the user's 4-digit discriminator
      * The discriminator is unique number to distinct one among all users with the same username.
      * The discriminator is randomly generated, but can be changed if the user has a nitro subscription.
+     * Migrated users from the old system will have a discriminator of "0".
+     * May become null after the system change is complete.
      *
-     * @return The user's 4-digit discriminator.
+     * @return The user's 4-digit discriminator, or "0" if the user is migrated to the new system.
+     * @deprecated This method will be removed once the system change is complete.
      */
+    @Deprecated
+    @Nullable
     public final String getDiscriminator() {
         return data.discriminator();
     }
 
     /**
-     * Gets the user's username and discriminator separated by a #
-     * This is unique across the discord platform, but may change.
-     * @return {@link User#getUsername()}#{@link User#getDiscriminator()}
+     * Returns whether the user is migrated to the new system or not.
+     *
+     * @return true if the user is migrated to the new system
+     */
+    private boolean isMigrated() {
+        return getDiscriminator() == null || getDiscriminator().equals("0");
```

Comment:
Let's make it simpler by checking whether globalName is present instead

Reply:
global_name its like the display_name [(a name non-unique to show)](https://github.com/discord/discord-api-docs/commit/6e9622bec117934aba362bb8c6b0389ef4c3f794#diff-39e6aeaafba5726f3fb409da21a6953dfb679480f09038a0d05f4dd93e89e074R46), this field its not a indicator if a user migrate or not.